sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -181,6 +190,7 @@ class RocksDB(
     try {
       if (loadedVersion != version) {
         closeDB()
+        fileManager = fileManager.deepCopy()
```

Comment:
Can we comment here why we want need to make a deep copy here?

Reply:
Added comment